patches/minecraft/net/minecraft/client/renderer/block/model/ItemTransforms.java.patch

```diff
@@ -1,26 +1,122 @@
++      this(p_111798_, p_111799_, p_111800_, p_111801_, p_111802_, p_111803_, p_111804_, p_111805_, com.google.common.collect.ImmutableMap.of());
++   }
++
++   public ItemTransforms(ItemTransform p_111798_, ItemTransform p_111799_, ItemTransform p_111800_, ItemTransform p_111801_, ItemTransform p_111802_, ItemTransform p_111803_, ItemTransform p_111804_, ItemTransform p_111805_,
++           com.google.common.collect.ImmutableMap<TransformType, ItemTransform> moddedTransforms) {
        this.f_111787_ = p_111798_;
        this.f_111788_ = p_111799_;
-@@ -47,6 +_,7 @@
+       this.f_111789_ = p_111800_;
+@@ -45,6 +_,7 @@
+       this.f_111792_ = p_111803_;
+       this.f_111793_ = p_111804_;
        this.f_111794_ = p_111805_;
++      this.moddedTransforms = moddedTransforms;
     }
  
-+   @Deprecated
     public ItemTransform m_111808_(ItemTransforms.TransformType p_111809_) {
-       switch(p_111809_) {
-       case THIRD_PERSON_LEFT_HAND:
+@@ -66,7 +_,7 @@
+       case FIXED:
+          return this.f_111794_;
+       default:
+-         return ItemTransform.f_111754_;
++         return moddedTransforms.getOrDefault( p_111809_, ItemTransform.f_111754_);
+       }
+    }
+ 
+@@ -94,7 +_,19 @@
+          ItemTransform itemtransform5 = this.m_111815_(p_111822_, jsonobject, "gui");
+          ItemTransform itemtransform6 = this.m_111815_(p_111822_, jsonobject, "ground");
+          ItemTransform itemtransform7 = this.m_111815_(p_111822_, jsonobject, "fixed");
+-         return new ItemTransforms(itemtransform1, itemtransform, itemtransform3, itemtransform2, itemtransform4, itemtransform5, itemtransform6, itemtransform7);
++
++         var builder = com.google.common.collect.ImmutableMap.<TransformType, ItemTransform>builder();
++         for(TransformType type : TransformType.values()) {
++            if (type.isModded()) {
++               var transform = this.m_111815_(p_111822_, jsonobject, type.getSerializeName());
++               if (transform != ItemTransform.f_111754_)
++               {
++                  builder.put(type, transform);
++               }
++            }
++         }
++
++         return new ItemTransforms(itemtransform1, itemtransform, itemtransform3, itemtransform2, itemtransform4, itemtransform5, itemtransform6, itemtransform7, builder.build());
+       }
+ 
+       private ItemTransform m_111815_(JsonDeserializationContext p_111816_, JsonObject p_111817_, String p_111818_) {
+@@ -103,19 +_,46 @@
+    }
+ 
+    @OnlyIn(Dist.CLIENT)
+-   public static enum TransformType {
+-      NONE,
+-      THIRD_PERSON_LEFT_HAND,
+-      THIRD_PERSON_RIGHT_HAND,
+-      FIRST_PERSON_LEFT_HAND,
+-      FIRST_PERSON_RIGHT_HAND,
+-      HEAD,
+-      GUI,
+-      GROUND,
+-      FIXED;
++   public static enum TransformType implements net.minecraftforge.common.IExtensibleEnum {
++      NONE("none"),
```

Comment:
These look to be pretty uniformly the lowercase of the enum identifier, so you could reduce patch size a lot by doing a zero arg constructor

private TransformType() {
  this.serializedName = this.name().toLowerCase(Locale.ROOT);
  isModded = false;
}

Reply:
Sadly no, as I just learned while writing the test mod:
```
      THIRD_PERSON_LEFT_HAND("third_person_lefthand"),
      THIRD_PERSON_RIGHT_HAND("third_person_righthand"),
      FIRST_PERSON_LEFT_HAND("first_person_lefthand"),
      FIRST_PERSON_RIGHT_HAND("first_person_righthand"),
```
the names don't quite match up.

Reply:
I assume having both a zero arg (for exceptions) and a 1 arg parameter won't save much in patch size.

I suppose you could could use a switch for the exceptions, but not convinced that would be cleaner either.